AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -106,3 +187,51 @@ def validate_variable_count(self, expected) -> None:
         result = helper.wait_for_condition(lambda: expected == row_count, WAIT_TIME_SEC_3)
 
         assert result, "Variable count did not match expected value."
+
+    def toggle_pin_variable_button(self, variable_type: str) -> None:
+        """
+        Function for clicking the pin toggle that appears for variables in the create variable list
+
+        param variable_type: The type of variable you want to pin to the top of the list
+
+        returns None
+        """
+        pinned_toggle_data = self.__find_pinned_toggle_control(variable_type)
+
+        assert pinned_toggle_data[1] is not None, "Failed to find variable pinning qt widget"
+
+        toggle_state_is_different = self.__toggle_the_pinned_control(pinned_toggle_data)
+
+        assert toggle_state_is_different, "Variable pin not successfully toggled"
```

Comment:
Find a pattern to eliminate that mystery number. I know that index 1 is the `pinned_toggle`. If you follow my advice of using a tuple over a dictionary, you can unpack the tuple when retrieving the values.

Then when you go to call the `__toggle_the_pinned_control` you can repack the tuple.

NIT: Let's remove the white space between each data retrieval and assert with whitespace between the two.

EG:

```suggestion
        variable_table_view, pinned_toggle = self.__find_pinned_toggle_control(variable_type)
        assert pinned_toggle is not None, "Failed to find variable pinning qt widget"

        toggle_state_is_different = self.__toggle_the_pinned_control((variable_table_view, pinned_toggle))
        assert toggle_state_is_different, "Variable pin not successfully toggled"
```

Reply:
fought through some typos while implementing this but I converted this over to the multiple return approach